src/cata_tiles.cpp

```diff
@@ -3389,4 +3392,50 @@ std::vector<options_manager::id_and_option> cata_tiles::build_renderer_list()
     return renderer_names.empty() ? default_renderer_names : renderer_names;
 }
 
+bool cata_tiles::save_screenshot( const std::string &file_path ) const
+{
+    std::string current_file_path;
+    if( file_path.empty() ) {
+        // check the current '<world>/screenshots' directory exists
+        std::stringstream map_directory;
+        map_directory << g->get_world_base_save_path() << "/screenshots/";
```

Comment:
This isn't particularly save. The world may not exits yet, in which case this would likely crash, but the function's doc doesn't mention that. At the same time, it accepts `file_path` which is always empty (there's no invocation with non-empty `file_path` yet). I'd move this snipped (everything after `if( file_path.empty() ) {`) elsewhere and pass it as a parameter to the function.